Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionSonicWallFirewall.yaml

```diff
@@ -0,0 +1,432 @@
+  | invoke _ASIM_ResolveDvcFQDN("")
+  | parse-kv AdditionalExtensions as (['gcat']:string, ['snpt']:string, ['dnpt']:string, ['susr']:string,['appName']:string, ['appcat']:string, ['appid']:string, ['sid']:string, ['catid']:string, ['ipscat']:string, ['ipspri']:string, ['spycat']:string, ['spypri']:string, ['fw_action']:string, ['dpi']:string, ['bid']:string, ['af_action']:string, ['af_polid']:string, ['af_policy']:string, ['af_type']:string, ['af_service']:string, ['af_object']:string, ['contentObject']:string, ['fileid']:string, ['uuid']:string) with (pair_delimiter=";", kv_delimiter="=")
+  | extend RequestURL_ = extract(@"(?:[.*;]+?)(?P<url>[a-zA-Z0-9_*.,}{&%$~:;\-=\/?[:space:]]+)(?:;fw_action)", 1, AdditionalExtensions)
+  | extend RequestURL_ = iif(RequestURL_ startswith "snpt" or RequestURL_ startswith "dnpt" or RequestURL_ startswith "appid" or RequestURL_ startswith "appName", extract(@"(?:\d;|.{1}\w.{1};)(?P<url>[a-zA-Z0-9_*.,}{&%$~:;\-=\/?[:space:]]+)", 1, RequestURL_), RequestURL_)
+  | extend RequestURL_ = iif(RequestURL_ matches regex @"^(.{2,6}=.{1,6})", extract(@"(?:\d;|.{1}\w.{1};)(?P<url>[a-zA-Z0-9_*.,}{&%$~:;\-=\/?[:space:]]+)", 1, RequestURL_), iif(RequestURL_ matches regex @"^\w=\d$", "", RequestURL_))
+  | lookup Actions on fw_action
+  // Sets the mandatory EventResult based on the DvcAction.
+  | extend EventResult = case(DvcAction == "Allow", "Success",
+                            DvcAction == "Management", "NA",
+                            DvcAction == "NA", "NA",
+                            "Failure"
+                            )
+  | extend LogMsgSeverity = case(LogSeverity == 10, "Emergency (0)",
+                                              LogSeverity == 9, "Alert (1)",
+                                              LogSeverity == 8, "Critical (2)",
+                                              LogSeverity == 7, "Error (3)",
+                                              LogSeverity == 6, "Warning (4)",
+                                              LogSeverity == 5, "Notice (5)",
+                                              LogSeverity == 4, "Info (6)/Debug (7)",
+                                              LogSeverity == 3, "Not Mapped (3)",
+                                              LogSeverity == 2, "Not Mapped (2)",
+                                              LogSeverity == 1, "Not Mapped (1)",
+                                              "Not Mapped"
+                                              )
+  | extend EventSeverity = case(tolong(LogSeverity) <= 4, "Informational",
+                                              tolong(LogSeverity) <= 6, "Low",
+                                              tolong(LogSeverity) <= 8, "Medium",
+                                              tolong(LogSeverity) > 8, "High",
+                                              ""
+                                              )
+  | extend HttpRequestMethod = case(RequestMethod == 0, "No Operation (0)",

```

Comment:
this field is not in Network Session schema

Reply:
Just so that I'm clear, is it supported to have fields that are not standard in the schema? I have many fields that are parsed from within AdditionalExtensions and fields renamed to meaningful field names from generic custom fields.

Reply:
I see now that I can use AdditionalFields to include additional details. I'm making the necessary changes to remove anything not in schema.

Reply:
sounds good